includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -363,6 +363,21 @@ public function process_subscription_payment( $amount, $renewal_order, $retry =
 				throw new WC_Stripe_Exception( print_r( $response, true ), $localized_message );
 			}
 
+			// TODO: Remove when SEPA is migrated to payment intents.
+			if ( 'stripe_sepa' !== $this->id ) {
+				$this->unlock_order_payment( $renewal_order );
+			}
+		} catch ( WC_Stripe_Exception $e ) {
+			WC_Stripe_Logger::log( 'Error: ' . $e->getMessage() );
+
+			do_action( 'wc_gateway_stripe_process_payment_error', $e, $renewal_order );
+
+			/* translators: error message */
+			$renewal_order->update_status( 'failed' );
+		}
+
+		try {
+
 			// Either the charge was successfully captured, or it requires further authentication.
 			if ( $is_authentication_required ) {
```

Comment:
If an exception is thrown in the top half, variables like `$response` and `$is_authentication_required` won't exist.

Do we need to add a `return;` in the catch statement above, or is it intended to still execute the charge requests even if the payment fails?

Reply:
Good catch Matt. I have added `return` in the first catch [92465a1](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3444/commits/92465a1de8e512e5e135521a1247b4fe6a1caed4).